classes/core/PKPApplication.inc.php

```diff
@@ -751,11 +751,11 @@ public static function getWorkflowTypeRoles() {
 	public static function getWorkflowStageName($stageId) {
 		AppLocale::requireComponents(LOCALE_COMPONENT_PKP_SUBMISSION, LOCALE_COMPONENT_APP_SUBMISSION);
 		switch ($stageId) {
-			case WORKFLOW_STAGE_ID_SUBMISSION: return __('submission.submission');
-			case WORKFLOW_STAGE_ID_INTERNAL_REVIEW: return __('workflow.review.internalReview');
-			case WORKFLOW_STAGE_ID_EXTERNAL_REVIEW: return __('workflow.review.externalReview');
-			case WORKFLOW_STAGE_ID_EDITING: return __('submission.editorial');
-			case WORKFLOW_STAGE_ID_PRODUCTION: return __('submission.production');
+			case WORKFLOW_STAGE_ID_SUBMISSION: return 'submission.submission';
+			case WORKFLOW_STAGE_ID_INTERNAL_REVIEW: return 'workflow.review.internalReview';
+			case WORKFLOW_STAGE_ID_EXTERNAL_REVIEW: return 'workflow.review.externalReview';
+			case WORKFLOW_STAGE_ID_EDITING: return 'submission.editorial';
+			case WORKFLOW_STAGE_ID_PRODUCTION: return 'submission.production';
 		}
 		throw new Exception('Name requested for an unrecognized stage id.');
 	}
```

Comment:
Check whether it makes sense (globally) to keep a map of constants and their names, and also, if here is the best place to keep such helper, maybe there's a workflow related class somewhere?

Reply:
Just a heads-up that this method is probably used in OMP and maybe OPS as well. Not a problem, just need to update where it's used in those applications.

Reply:
Thanks for the reminder! I wouldn't remember to look for dependencies in the other repositories... Hopefully I didn't find references for it =]

As a side note: What about breaking the pkp-lib in pieces and serving them with packages? It would increase the visibility of such breaking changes and allow for "incremental updates". The trade-off is the dependency hell.

Reply:
It would take quite a bit of work to find good cleave points for pkp-lib, but I'm open to the idea!